connectors/src/connectors/microsoft/temporal/file.ts

```diff
@@ -282,17 +214,73 @@ export async function syncOneFile({
     name: file.name ?? "",
     parentInternalId,
     mimeType: file.file.mimeType ?? "",
-    lastUpsertedTs:
-      isInSizeRange && upsertTimestampMs ? new Date(upsertTimestampMs) : null,
   };
 
+  if (result.isOk() && result.value) {
+    // In case of table, result is ok and result.value is null, no document to upsert
+    const documentSection = result.value;
+    const documentLength = sectionLength(documentSection);
+    const isInSizeRange = documentLength > 0 && documentLength < maxDocumentLen;
+
+    if (isInSizeRange) {
+      localLogger.info({ documentSection }, "Document section");
+      const content = await renderDocumentTitleAndContent({
+        dataSourceConfig,
+        title: file.name ?? null,
+        updatedAt,
+        createdAt,
+        lastEditor: file.lastModifiedBy?.user
+          ? file.lastModifiedBy.user.displayName ?? undefined
+          : undefined,
+        content: documentSection,
+      });
+
+      const upsertTimestampMs = updatedAt ? updatedAt.getTime() : undefined;
+
+      const parents = await getParents({
+        connectorId,
+        internalId: documentId,
+        parentInternalId,
+        startSyncTs,
+      });
+      parents.reverse();
+
+      await upsertToDatasource({
+        dataSourceConfig,
+        documentId,
+        documentContent: content,
+        documentUrl: file.webUrl ?? undefined,
+        timestampMs: upsertTimestampMs,
+        tags,
+        parents: parents,
+        upsertContext: {
+          sync_type: isBatchSync ? "batch" : "incremental",
+        },
+        async: true,
+      });
+
+      resourceBlob.lastUpsertedTs = upsertTimestampMs
+        ? new Date(upsertTimestampMs)
+        : null;
+    } else {
+      localLogger.info(
+        {
+          documentLen: documentLength,
+        },
+        `Document is empty or too big to be upserted (marking as synced without upserting)`
+      );
+
+      result = new Err(new Error("document_too_big_or_empty"));
+    }
+  }
+
   if (fileResource) {
     await fileResource.update(resourceBlob);
```

Comment:
So we don't differentiate successfully imported than the one that failed to be sync. This seems to be an issue as those docs will show up in the the assistant builder whereas they don't have any data.